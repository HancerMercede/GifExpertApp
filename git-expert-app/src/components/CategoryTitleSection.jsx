import { PropTypes } from "prop-types";
import styles from "./CategoryTitleSection.module.css";

export const CategoryTitleSection = ({ category }) => {
  return (
    <div className={styles.container}>
      <h1 className={styles.categoryTitle}>{category}</h1>
    </div>
  );
};

CategoryTitleSection.propTypes = {
  category: PropTypes.string.isRequired,
};
