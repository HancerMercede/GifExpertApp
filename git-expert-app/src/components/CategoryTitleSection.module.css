.container {
    display: flex;
    justify-content: center;
    color: white;
    margin: 5px 0px 5px 0px;
}

.categoryTitle {
    text-shadow: 2px 4px 5px rgba(204, 236, 85, 0.5);
    font-family: 'Trebuchet MS', 'Lucida Sans Unicode', 'Lucida Grande', 'Lucida Sans', Arial, sans-serif;
}