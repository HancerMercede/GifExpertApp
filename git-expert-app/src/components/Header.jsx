import { PropTypes } from "prop-types";
import styles from "./Header.module.css";

export const Header = ({ AppTitle }) => {
  return (
    <div className={styles.container}>
      <h1 className={styles.AppTitle}>{AppTitle}</h1>
    </div>
  );
};

Header.propTypes = {
  AppTitle: PropTypes.string.isRequired,
};
