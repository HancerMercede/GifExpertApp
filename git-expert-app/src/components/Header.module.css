@import url('https://fonts.googleapis.com/css2?family=Tilt+Prism&display=swap');

.container {
    display: flex;
    justify-content: center;
}

.AppTitle {
    font-size: 4rem;
    font-family: 'Tilt Prism', cursive;
    color: rgb(241, 196, 15);
    text-shadow: 10px 8px 8px rgba(215, 228, 171, 0.2);
}