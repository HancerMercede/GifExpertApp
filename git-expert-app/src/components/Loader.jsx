import { ImSpinner2 } from "react-icons/im";
import styles from "./Loader.module.css";

export const Loader = () => {
  return (
    <div className={styles.spinner}>
      <ImSpinner2 className={styles.spinning} size={40} />
    </div>
  );
};
