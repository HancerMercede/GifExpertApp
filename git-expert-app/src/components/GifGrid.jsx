import { useFetchGifs } from "../hooks/useFetchGifs";
import { CategoryTitleSection } from "./CategoryTitleSection";
import { GifItem } from "./GifItem";
import { PropTypes } from "prop-types";

export const GifGrid = ({ category }) => {
  const { images, isLoading } = useFetchGifs(category);

  return (
    <>
      <CategoryTitleSection category={category} />

      {isLoading && <h2>Loading...</h2>}
      <div className="card-grid">
        {images.map((image) => (
          <GifItem key={image.id} {...image} />
        ))}
      </div>
    </>
  );
};

GifGrid.propTypes = {
  category: PropTypes.string.isRequired,
};
