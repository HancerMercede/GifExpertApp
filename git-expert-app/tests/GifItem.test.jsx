/* eslint-disable no-undef */
import {render, screen} from '@testing-library/react';
import {GifItem} from '../src/components/GifItem';


describe('Pruebas en <GifItem />',() => {

   const title= 'Mokoto Kusanagi';
   const url = 'http://localhost:5173/';
 
   test('Debe hacer match con el snapshop', () => {
     
      const {container} = render(<GifItem  title={title} url={url}/>);

      expect(container).toMatchSnapshot()

   });


   test('Debe de mostrar la imagen con el URL y el ALT indicado', () => {
    
       render(<GifItem  title={title} url={url}/>);
       // screen.debug();
      // expect(screen.getByRole('img').src).toBe(url);
       const {src, alt } = screen.getByRole('img');
       expect(src).toBe(url);
       expect(alt).toBe(title);

   })

   test('Debe de mostrar el titulo en el componente',()=>{
      render(<GifItem  title={title} url={url}/>);
      expect(screen.getAllByText(title)).toBeTruthy();
   })

});