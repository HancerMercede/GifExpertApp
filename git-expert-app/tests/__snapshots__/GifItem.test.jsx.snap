// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifItem /> Debe hacer match con el snapshop 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Mokoto Kusanagi"
      src="http://localhost:5173/"
    />
    <p>
      Mokoto Kusanagi
    </p>
  </div>
</div>
`;
